### PR TITLE
fix: Fix skip reconciliation for Manifest

### LINKED ISSUE #937: Skip labels get removed or ignored
**Description**
The functionality to skip Manifest reconciliation for SRE debugging purposes is not working as expected. The label of  `declarative.kyma-project.io/skip-reconciliation: "true"` actually gets removed again once it has been edited, but it should not.

**Acceptance Criteria**
* Delete [this](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/declarative/v2/options.go#L25)
    * Reuse the label from [v1beta2 API](https://github.com/kyma-project/lifecycle-manager/blob/4851440431fe021751fccae7a2d4a56e5452cd4e/api/v1beta2/operator_labels.go#L31)
* Manifest label `operator.kyma-project.io/skip-reconciliation: "true"` should be excluded from Kyma reconciliation
    * Fetch ManifestCR
    * If ManifestCR contains this label, ManifestCR labels should not be updated
    * Test Performance + Report
* Implement integration tests fro Kyma controller
    * Implement [here](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/controller/kyma_controller/manifest_test.go)
    * Check if label is still existing after one or multiple Kyma reconciliation loops in corresponding ManifestCR
        * Use `Consistently()` function, not `Eventually()`
* Create Follow-Up issue to test skip-reconciliation functionality of Manifest Controller

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Added an integration test for the skip-label on the Manifest
- Renamed the `declarative.kyma-project.io/skip-reconciliation` to `operator.kyma-project.io/skip-reconciliation`

**Related issue(s)**
Resolves #937

### CODE REVIEW

File: api/v1beta2/kyma_types.go [hunk 1]
```diff
@@ -416,7 +417,7 @@ func (kyma *Kyma) HasSyncLabelEnabled() bool {
 }
 
 func (kyma *Kyma) SkipReconciliation() bool {
-	skip, found := kyma.Labels[SkipReconcileLabel]
+	skip, found := kyma.Labels[v2.DefaultSkipReconcileLabel]
```
Comment (lindnerby) [inline api/v1beta2/kyma_types.go:420]: Btw why is this "default", can be named SkipReconileLabel right?

File: internal/declarative/v2/options.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ const (
 	FinalizerDefault          = "declarative.kyma-project.io/finalizer"
 	FieldOwnerDefault         = "declarative.kyma-project.io/applier"
 	EventRecorderDefault      = "declarative.kyma-project.io/events"
-	DefaultSkipReconcileLabel = "declarative.kyma-project.io/skip-reconciliation"
+	DefaultSkipReconcileLabel = "operator.kyma-project.io/skip-reconciliation"
```
Comment (lindnerby) [inline internal/declarative/v2/options.go:25]: Let's rename it :)

**Review by lindnerby — CHANGES_REQUESTED** [review]
Just a small renaming request, rest looks good.

Did you check if we have any documentation for this label to rename as well?
If there is no documentation we should probably create an issue to create some.

**Comment by nesmabadr** [comment]
Documentation issue: https://github.com/kyma-project/lifecycle-manager/issues/973

**Review by lindnerby — APPROVED** [review]

**Pull request merged**